tc/sgx/trusted_worker_manager/enclave/CMakeWorkloads.txt

```diff
@@ -0,0 +1,58 @@
+# Copyright 2020 Intel Corporation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# Create list of supported workloads
+LIST(APPEND SUPPORTED_WORKLOADS_LIST "echo-result")
+LIST(APPEND SUPPORTED_WORKLOADS_LIST "heart-disease-eval")
+LIST(APPEND SUPPORTED_WORKLOADS_LIST "inside-out-eval")
+LIST(APPEND SUPPORTED_WORKLOADS_LIST "simple-wallet")
+
+# A new workload cna be added by using below syntax
+# LIST(SUPPORTED_WORKLOADS_LIST "<workload_id>")
+
+# Create list of supported workload library names
+LIST(APPEND SUPPORTED_WORKLOAD_LIBRARY_LIST "echo_wo")
+LIST(APPEND SUPPORTED_WORKLOAD_LIBRARY_LIST "heart_disease_eval")
+LIST(APPEND SUPPORTED_WORKLOAD_LIBRARY_LIST "inside_out_eval")
+LIST(APPEND SUPPORTED_WORKLOAD_LIBRARY_LIST "simple_wallet")
+
+# A new workload library can be added by using below syntax
+# LIST(APPEND SUPPORTED_WORKLOAD_LIBRARY_LIST "<workload_lib_name>")
+
+SET(WORKLOAD_LIST "$ENV{WORKLOADS}")
+# Create workload library list based on workloads passed
+IF("${WORKLOAD_LIST}" STREQUAL "all" OR "${WORKLOAD_LIST} " STREQUAL " ")
+    SET(WORKLOAD_LIB_LIST ${SUPPORTED_WORKLOAD_LIBRARY_LIST})
+ELSE()
+    FOREACH(workload ${WORKLOAD_LIST})
+        # Check if user provided workload is present in supported workload list
+        # If not found, return -1
+        LIST(FIND SUPPORTED_WORKLOADS_LIST ${workload} index)
+        IF(NOT ${index} EQUAL -1)
+            # Get workload library corresponding to workload identifier
+            LIST(GET SUPPORTED_WORKLOAD_LIBRARY_LIST ${index} workload_lib)
+            LIST(APPEND WORKLOAD_LIB_LIST ${workload_lib})
+        ELSE()
+            MESSAGE(FATAL_ERROR,
+                "Unable to find workload library corresponding to parsed workload")
+        ENDIF()
+    ENDFOREACH(workload)
+ENDIF()
+
+MACRO(LINK_ENCLAVE_WORKLOAD_LIBRARIES)
+    FOREACH(workload_lib ${WORKLOAD_LIB_LIST})
+        TARGET_LINK_LIBRARIES(
+            ${PROJECT_NAME} -Wl,--whole-archive -l${workload_lib} -Wl,--no-whole-archive)
+    ENDFOREACH(workload_lib)
+ENDMACRO()
```

Comment:
Please update the relevant documentation below for how to add a new workload https://github.com/hyperledger/avalon/tree/master/docs/workload-tutorial

Reply:
Done